src/main/java/uk/gov/hmcts/reform/civil/service/user/UserInformationService.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.hmcts.reform.civil.service.user;
+
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.civil.exceptions.CaseNotFoundException;
+import uk.gov.hmcts.reform.civil.service.CoreCaseUserService;
+import uk.gov.hmcts.reform.civil.service.UserService;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class UserInformationService {
+
+    Logger log = LoggerFactory.getLogger(UserInformationService.class);
+
+    private final UserService userService;
+    private final CoreCaseUserService coreCaseUserService;
+
+    public List<String> getUserCaseRoles(String caseId, String authorization) {
+        try {
+            UserInfo userInfo = userService.getUserInfo(authorization);
+            List<String> roles = coreCaseUserService.getUserCaseRoles(
+                caseId,
+                userInfo.getUid()
+            );
+            return roles;
+        } catch (Exception e) {
+            log.error(String.format("No case found for %s", caseId));
```

Comment:
you could also hit this catch if you try access a case that youre not on, make sure youre logging the error message here otherwise itll be really difficult to see when this goes wrong for the user not being on the case

Reply:
or if this is the camunda exception that doesnt provide much info, then maybe just tweak the error message to also account for user not being on case too